### PR TITLE
fix(Dropdown): Allow clear on Enter/Space press

### LINKED ISSUE #26258: [Bug]: Clearable dropdown > clear button is not invoked by Enter/Space key
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 12.27 GB / 31.74 GB
  Browsers:
    Edge: Spartan (44.22621.963.0), Chromium (108.0.1462.76)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/s/vqmboh?module=/example.js

### Bug Description

Repro steps:
1. Select any value in clearable dropdown
2. navigate to close button
3. press enter/space

## Actual Behavior
nothing happen

## Expected Behavior
onclick action should be executed and dropdown cleared... 

the same repro on docsite:
https://fluentsite.z22.web.core.windows.net/0.65.0/components/dropdown/definition?showCode=false&showRtl=false&showTransparent=false&showVariables=false#types-clearable

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #26563: [Bug]: Dropdown clear indicator should not be aria-hidden
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz
    Memory: 6.28 GB / 31.71 GB
  Browsers:
    Edge: Spartan (44.22621.1105.0), Chromium (109.0.1518.61)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://fluentsite.z22.web.core.windows.net/0.65.0/components/dropdown/definition

## Actual Behavior
When you request "show numbers" with voice access, there is no number for the clear indicator because it has aria-hidden = true. 

## Expected Behavior
If aria-hidden is removed from the clear indicator, then show numbers does successfully show a number for the clear indicator
issue can be observed on fluent site when asking for show numbers

### Logs

_No response_

### Requested priority

Blocking

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by jurokapsiar**
## Previous Behavior

clearable indicator was hidden and not keyboard accessible

## New Behavior

remove aria-hidden, add key down handlers for Enter and Space

## Related Issue(s)

Fixes #26563 
Fixes #26258

### CODE REVIEW

**Review by layershifter — APPROVED** [review]

File: packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -1744,14 +1744,22 @@ export const Dropdown = (React.forwardRef<HTMLDivElement, DropdownProps>((props,
...
                         onClick: (e: React.SyntheticEvent<HTMLElement>) => {
                           _.invoke(predefinedProps, 'onClick', e);
                           handleClear(e);
                         },
+                        onKeyDown: (e: React.KeyboardEvent<HTMLElement>) => {
+                          _.invoke(predefinedProps, 'onKeyDown', e);
+                          const keyCode = getCode(e);
+                          if (!search && (keyCode === keyboardKey.Enter || keyCode === SpacebarKey)) {
+                            handleClear(e);
+                            e.stopPropagation();
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx:1759]: would `stopPropagation` be a breaking change? if any user relies on that event, after this change they will no longer get it

Comment (jurokapsiar) [reply packages/fluentui/react-northstar/src/components/Dropdown/Dropdown.tsx]: looks like preventDefault is enough, so I removed stopPropagation.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c625afa508c0dead93ea2d3cb698b41dc67ec187'>c625afa508c0dead93ea2d3cb698b41dc67ec187</a></sub>
<div id="bundleSizeComment9423"/>

**Review by ling1726 — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 FluentUI-v0 No visual regressions between this PR and main
<div id="vrtCommentFluentUI-v0"/>

**Comment by 1hw7** [comment]
Hi! Thanks for doing this! quick question, I'm wondering when your team does releases and when we should expect to see the change?

**Pull request merged**